lib/services/search.js

```diff
@@ -0,0 +1,387 @@
+      return bluebird.reject(error);
+    });
+}
+
+
+/**
+ * Index an Elasticsearch document
+ *
+ * @param {string} index
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @param {object} source
+ * @returns {Promise}
+ */
+function put(index, type, ref, source) {
+  return client.index({
+    index: index,
+    type: type,
+    id: ref,
+    body: source
+  }).then(function (resp) {
+    log('info', resp);
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Delete an Elasticsearch document
+ *
+ * @param {string} index
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @returns {Promise}
+ */
+function del(index, type, ref) {
+  return client.delete({
+    index: index,
+    type: type,
+    id: ref
+  }).then(function (resp) {
+    log('info', resp);
+    return resp;
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Perform multiple index operations
+ *
+ * @param {Array} ops
+ * @returns {Promise}
+ */
+function batch(ops) {
+  return client.bulk({
+    body: ops
+  }).then(function (resp) {
+    if (resp && resp.errors === true) {
+      let str = ['Client.bulk errored on ' + util.inspect(ops, {depth: 10})];
+
+      return bluebird.reject(new Error(str));
+    }
+  });
+}
+
+/**
+ * Create the ES Client or an empty object
+ *
+ * @param {Object} overrideClient
+ */
+function setup(overrideClient) {
+  if (!module.exports.endpoint && !overrideClient) {
+    throw new Error('No Elastic endpoint or client override');
+  }
+
+  // Set the exported client
+  module.exports.client = client = module.exports.endpoint && !overrideClient ? new elastic.Client(serverConfig) : _.assign(module.exports.client, overrideClient);
+}
+
+/**
+ * Sets the index version. Index version is required
+ * to begin with '_v' to denote a version of the index.
+ *
+ * @param {string} suffix
+ */
+function setIndexVersion(suffix) {
+  if (_.isString(suffix) && _.includes(suffix, '_v')) {
+    indexVersion = suffix;
+  } else {
+    throw new Error('Index version required to be a string beginning with "_v"');
+  }
+}
+
+module.exports.setup = setup;
+module.exports.setIndexVersion = setIndexVersion;
+module.exports.endpoint = endpoint;
+module.exports.healthCheck = healthCheck;
+module.exports.initIndex = initIndex;
+module.exports.initAlias = initAlias;
+module.exports.createIndexIfNone = createIndexIfNone;
+module.exports.createAliasIfNone = createAliasIfNone;
+module.exports.existsAlias = existsAlias;
+module.exports.existsIndex = existsIndex;
+module.exports.initMapping = initMapping;
+module.exports.existsMapping = existsMapping;
+module.exports.createMappingIfNone = createMappingIfNone;
+module.exports.createIndexName = createIndexName;
```

Comment:
why is this being exported? and why are we calling it during a batch op in sites? https://github.com/nymag/sites/blob/search-service/services/search/index.js#L267